src/broker/controller-dbus.c

```diff
@@ -662,3 +693,42 @@ int controller_dbus_send_environment(Controller *controller, const char * const
 
         return 0;
 }
+
+/**
+ * controller_dbus_send_reload() - XXX
+ */
+int controller_dbus_send_reload(Controller *controller, uint32_t serial) {
+        static const CDVarType type[] = {
+                C_DVAR_T_INIT(
+                        CONTROLLER_T_MESSAGE(
+                                C_DVAR_T_TUPLE0
+                        )
+                )
+        };
+        _c_cleanup_(c_dvar_deinit) CDVar var = C_DVAR_INIT;
+        _c_cleanup_(message_unrefp) Message *message = NULL;
+        size_t n_data;
+        void *data;
+        int r;
+
+        c_dvar_begin_write(&var, type, 1);
+        c_dvar_write(&var, "((yyyyuu[(y<o>)(y<s>)(y<s>)])())",
+                     c_dvar_is_big_endian(&var) ? 'B' : 'l', DBUS_MESSAGE_TYPE_METHOD_CALL, 0, 1, 0, serial,
+                     DBUS_MESSAGE_FIELD_PATH, c_dvar_type_o, "/org/bus1/DBus/Launcher",
+                     DBUS_MESSAGE_FIELD_INTERFACE, c_dvar_type_s, "org.bus1.DBus.Launcher",
+                     DBUS_MESSAGE_FIELD_MEMBER, c_dvar_type_s, "ReloadConfig");
+
+        r = c_dvar_end_write(&var, &data, &n_data);
+        if (r)
+                return error_origin(r);
+
+        r = message_new_outgoing(&message, data, n_data);
+        if (r)
+                return error_fold(r);
```

Comment:
This leaks `data`, doesn't it? Shouldn't it be marked as `_c_cleanup_(c_freep)` and NULL'ified after this call?

Reply:
ugh... yeah, and this is wrong everywhere. Fixed it.